kernel/kernel-api/src/main/java/io/delta/kernel/internal/metrics/Timer.java

```diff
@@ -86,6 +86,29 @@ public <T> T timeCallable(Callable<T> callable) throws Exception {
     }
   }
 
+  /**
+   * Times an operation that can throw a specific checked exception type.
+   *
+   * @param <T> The return type
+   * @param <E> The exception type that can be thrown
+   * @return The result of the operation
+   * @throws E The exception thrown by the operation
+   */
+  @FunctionalInterface
+  public interface ThrowingSupplier<T, E extends Exception> {
+    T get() throws E;
+  }
+
+  @SuppressWarnings("unchecked")
+  public <T, E extends Exception> T timeChecked(ThrowingSupplier<T, E> operation) throws E {
```

Comment:
I want to be able to time a method that throws a checked exception E, without having to catch the aprent Exception type. See:

https://github.com/delta-io/delta/blob/9c389562b5faaa1affa774d0fec50f2020ffbee6/unity/src/main/java/io/delta/unity/utils/OperationTimer.java#L33